### PR TITLE
Stop relying on external constants

### LINKED ISSUE #6732: HTMLEditorConfigTest & TinyMCEConfig relies on ADMIN_THIRDPARTY_DIR which isn't defined
The `HTMLEditorConfigTest` and `TinyMCEConfig` rely on the `ADMIN_THIRDPARTY_DIR` which is defined by the admin module but not in framework itself.

Either we need to remove these assertions or find a way to more loosely couple to admin.

### PR DESCRIPTION
**Pull request by dhensby**
fixes #6732

### CODE REVIEW

**Comment by sminnee** [comment]
The broader solution to this is discussed in #6243. I wonder if it would be a good time to start solving that?

**Comment by sminnee** [comment]
In particular, instead of this:

```php
Director::absoluteURL(ltrim($this->getAdminPath() . '/client/dist/styles/editor.css', '/'));
```

We could have

```php
Injector::inst()->get(ResourceLocator::class)->getResource('silverstripe/admin', 'client/dist/styles/editor.css')->getURL();
```

Or:

```php
Injector::inst()->get(ResourceLocator::class)->getResourceURL('silverstripe/admin', 'client/dist/styles/editor.css');
```

**Comment by dhensby** [comment]
Yes - I'd discussed this with @tractorcow when I was in Auckland. This was a short-term patch to get the tests running when framework was installed standalone (though clearly they don't all pass with this patch).

**Comment by tractorcow** [comment]
We did add the experimental API https://github.com/silverstripe/silverstripe-framework/issues/6733.

The question is, is it safe to rely on this, or is it time to make this api non-experimental and formalise it?

**Comment by sminnee** [comment]
OK I'm happy with this as a quick-fix; we can pick up the package resource issue after this is merged.

**Comment by tractorcow** [comment]
Given that we've merged https://github.com/silverstripe/silverstripe-framework/pull/6733, should we merge it using this API?

**Comment by tractorcow** [comment]
Unit tests break if admin isn't installed.

File: src/Forms/HTMLEditor/TinyMCEConfig.php [hunk 1]
```diff
@@ -676,4 +679,25 @@ public static function get_tinymce_lang()
...
+    /**
+     * @return string|false
+     */
+    public function getAdminPath()
+    {
+        $module = ModuleLoader::instance()->getManifest()->getModule('silverstripe/admin');
+        if ($module) {
+            return $module->getRelativePath();
+        }
+        return false;
```
Comment (tractorcow) [inline src/Forms/HTMLEditor/TinyMCEConfig.php]: This return false isn't handled in getTinyMCEPath().

**Comment by sminnee** [comment]
@dhensby this is the failure

```
1) SilverStripe\Forms\Tests\HTMLEditor\HTMLEditorConfigTest::testPluginCompression
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'test/thirdparty/tinymce/tiny_mce_gzip.php?js=1&plugins=plugin4,plugin5&themes=modern&languages=es&diskcache=true&src=true'
+'silverstripe-admin/thirdparty/tinymce/tiny_mce_gzip.php?js=1&plugins=plugin4,plugin5&themes=modern&languages=es&diskcache=true&src=true'
/home/travis/build/silverstripe/silverstripe-framework/tests/php/Forms/HTMLEditor/HTMLEditorConfigTest.php:132
/home/travis/build/silverstripe/silverstripe-framework/vendor/phpunit/phpunit/phpunit:52
``

**Comment by unclecheese** [comment]
I've fixed this up and submitted as a new PR https://github.com/silverstripe/silverstripe-framework/pull/6876

**Pull request closed without merging**